backend/tracim_backend/lib/collaborative_document_edition/collaboration_document_edition.py

```diff
@@ -0,0 +1,124 @@
+            mimetype = xml_app.get("name")
+            # INFO - G.M - 2019-07-17 - this list return also a non mimetype type,
+            # we should not add him.
+            if mimetype == "Capabilities":
+                continue
+            for xml_action in xml_app:
+                extension = xml_action.get("ext")
+                url_source = xml_action.get("urlsrc")
+                associated_action = xml_action.get("name")
+                supported_collabora_file.append(
+                    CollaborativeDocumentEditionFileType(
+                        mimetype=mimetype,
+                        extension=extension,
+                        associated_action=associated_action,
+                        url_source=url_source,
+                    )
+                )
+        return supported_collabora_file
+
+    def get_collaboration_document_edition_config(self) -> CollaborativeDocumentEditionConfig:
+        return CollaborativeDocumentEditionConfig(
+            software=self._config.COLLABORATIVE_DOCUMENT_EDITION__SOFTWARE,
+            supported_file_types=self.discover(),
+        )
+
+    def get_file_template_list(self) -> FileTemplateList:
+        return FileTemplateList(file_templates=self._get_template_list())
+
+    def _has_template_extension(self, template_name: str, extension_list: typing.List[str]):
+        for extension in extension_list:
+            if template_name.endswith(extension):
+                return True
+        return False
+
+    def _get_template_list(self) -> typing.List[str]:
+        template_list = []
+        try:
+            is_dir_exist(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_dir_readable(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+        except (NotADirectoryError) as exc:
+            raise FileTemplateNotAvailable from exc
+        for filename in os.listdir(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR):
+            if not isfile(
+                join(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR, filename)
+            ):
+                continue
+            template_list.append(filename)
+        return template_list
+
+    def _get_file_template_path(self, template_filename) -> str:
+        template_path = "{}/{}".format(
+            self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR, template_filename
+        )
+        try:
+            is_dir_exist(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_dir_readable(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_file_exist(template_path)
+            is_file_readable(template_path)
+        except (NotADirectoryError, NotAFileError, NotReadableFile) as exc:
+            raise FileTemplateNotAvailable from exc
+        return template_path
+
+    def check_template_available(self, template_filename) -> None:
```

Comment:
typing

Reply:
by the way. what does this method do? It does not return a value, so... so what? Miss a comment, I think

Reply:
done

Reply:
done.